Runtime/Client/LootLockerLifecycleManager.cs

```diff
@@ -0,0 +1,968 @@
+        }
+#endif
+
+        #endregion
+
+        #region Service Management
+
+        private readonly Dictionary<Type, ILootLockerService> _services = new Dictionary<Type, ILootLockerService>();
+        private readonly List<ILootLockerService> _initializationOrder = new List<ILootLockerService>();
+        private readonly List<Type> _serviceInitializationOrder = new List<Type>
+        {
+            // Define the initialization order here
+            typeof(RateLimiter), // Rate limiter first (used by HTTP client)
+            typeof(LootLockerHTTPClient),         // HTTP client second
+            typeof(LootLockerEventSystem),       // Events system third
+#if LOOTLOCKER_ENABLE_PRESENCE
+            typeof(LootLockerPresenceManager)     // Presence manager last (depends on HTTP)
+#endif
+        };
+        private bool _isInitialized = false;
+        private bool _serviceHealthMonitoringEnabled = true;
+        private Coroutine _healthMonitorCoroutine = null;
+        private static LifecycleManagerState _state = LifecycleManagerState.Ready;
+        private readonly object _serviceLock = new object();
+
+        /// <summary>
+        /// Register a service to be managed by the lifecycle manager.
+        /// Service is immediately initialized upon registration.
+        /// </summary>
+        public static void RegisterService<T>(T service) where T : class, ILootLockerService
+        {
+            var instance = Instance;
+            instance._RegisterServiceAndInitialize(service);
+        }
+
+        /// <summary>
+        /// Create and register a MonoBehaviour service component to be managed by the lifecycle manager.
+        /// Service is immediately initialized upon registration.
+        /// </summary>
+        public static T RegisterService<T>() where T : MonoBehaviour, ILootLockerService
+        {
+            var instance = Instance;
+            var service = instance.gameObject.AddComponent<T>();
+            instance._RegisterServiceAndInitialize(service);
+            return service;
+        }
+
+        /// <summary>
+        /// Get a service. The LifecycleManager auto-initializes on first access if needed.
+        /// </summary>
+        public static T GetService<T>() where T : class, ILootLockerService
+        {
+            if (_state == LifecycleManagerState.Quitting || _state == LifecycleManagerState.Resetting)
+            {
+                LootLockerLogger.Log($"Access of service {typeof(T).Name} during {_state.ToString().ToLower()} was requested but denied", LootLockerLogger.LogLevel.Debug);
+                return null;
+            }
+            
+            // CRITICAL: Prevent circular dependency during initialization
+            if (_state == LifecycleManagerState.Initializing)
+            {
+                LootLockerLogger.Log($"Service {typeof(T).Name} requested during LifecycleManager initialization - this could cause deadlock. Returning null.", LootLockerLogger.LogLevel.Warning);
```

Comment:
```suggestion
                LootLockerLogger.Log($"Service {typeof(T).Name} requested during LifecycleManager initialization - this could cause deadlock. Returning null.", LootLockerLogger.LogLevel.Info);
```

Reply:
Fixed